frmts/esric/esric_dataset.cpp

```diff
@@ -291,32 +312,156 @@ CPLErr ECDataset::Initialize(CPLXMLNode *CacheInfo)
     return error;
 }
 
+CPLErr ECDataset::InitializeFromJSON(const CPLJSONObject &oRoot)
+{
+    CPLErr error = CE_None;
+    try
+    {
+        auto format = oRoot.GetString("storageInfo/storageFormat");
+        isV2 = EQUAL(format.c_str(), "esriMapCacheStorageModeCompactV2");
+        if (!isV2)
+            throw CPLString("Not recognized as esri V2 bundled cache");
+        if (BSZ != oRoot.GetInteger("storageInfo/packetSize"))
+            throw CPLString("Only PacketSize of 128 is supported");
+
+        TSZ = oRoot.GetInteger("tileInfo/rows");
+        if (TSZ != oRoot.GetInteger("tileInfo/cols"))
+            throw CPLString("Non-square tiles are not supported");
+
+        auto oLODs = oRoot.GetArray("tileInfo/lods");
+        double res = 0;
+        // we need to skip levels that don't have bundle files
+        int minLOD = oRoot.GetInteger("minLOD");
+        int maxLOD = oRoot.GetInteger("maxLOD");
+        int level = 0;
+        for (const auto &oLOD : oLODs)
+        {
+            res = oLOD.GetDouble("resolution");
+            if (!(res > 0))
+                throw CPLString("Can't parse resolution for LOD");
+            level = oLOD.GetInteger("level");
+            if (level >= minLOD && level <= maxLOD)
+            {
+                resolutions.push_back(res);
+            }
+        }
+        sort(resolutions.begin(), resolutions.end());
+        if (resolutions.empty())
+            throw CPLString("Can't parse lods");
+
+        if (OGRERR_NONE !=
+            oSRS.SetFromUserInput(CPLString().Printf(
+                "ESRI:%d", oRoot.GetInteger("spatialReference/wkid"))))
```

Comment:
I'm not sure about this particular format, but with other Esri stuff (like FileGDB), you want to use the ESRI: namespace only for codes > 32767. For codes <= 32767, you have to use the EPSG: namespace

Reply:
Indeed, the previous test should work, according to the [documentation](https://github.com/Esri/raster-tiles-compactcache/blob/master/CompactCacheV2.md).

Reply:
@lucianpls u32lat(header + 4) seems to contain zero for the tpkx format, however we could leave it in for the xml based configurations.

Reply:
Bundles should be all the same format, this is likely a bug somewhere. How was the tile package generated?

Reply:
@lucianpls I got the files from the client, probably made with some ESRI tools. But you could also check [this example](https://github.com/Esri/tile-package-spec/tree/master) where that field is also zero:
![image](https://github.com/OSGeo/gdal/assets/1408872/835a9730-699a-4f56-9a69-4416e5dd85ca)

What do you mean by "Bundles should be all the same format". There might probably be slight differences between the implementation applications (like in the .tpkx case). Where can we find the official documentation about that?

Reply:
> I'm not sure about this particular format, but with other Esri stuff (like FileGDB), you want to use the ESRI: namespace only for codes > 32767. For codes <= 32767, you have to use the EPSG: namespace

@rouault The corresponding documentation can be found [here](https://github.com/Esri/tile-package-spec/blob/master/docs/root.md#spatialreference). As what I can see in the root.json file, the wkid corresponds to the ESRI id (102100), while the LatestWkid contains the EPSG code (3857).

Reply:
@rouault Thanks for the hint, that makes sense. I'll incorporate that in the code.

Reply:
> @lucianpls I got the files from the client, probably made with some ESRI tools. But you could also check [this example](https://github.com/Esri/tile-package-spec/tree/master) where that field is also zero: ![image](https://user-images.githubusercontent.com/1408872/240174472-835a9730-699a-4f56-9a69-4416e5dd85ca.png)
> 
> What do you mean by "Bundles should be all the same format". There might probably be slight differences between the implementation applications (like in the .tpkx case). Where can we find the official documentation about that?

Indeed, turns out esri code for building bundles is not consistent. It is best to leave the check the way you had it to start with, checking value at offset 60 for both tpkx and normal bundled cache.